hivemq-edge/src/frontend/src/modules/ProtocolAdapters/components/drawers/ExportDrawer.tsx

```diff
@@ -0,0 +1,222 @@
+import { FC, useEffect, useState } from 'react'
+import { useNavigate, useParams } from 'react-router-dom'
+import { useTranslation } from 'react-i18next'
+import { Controller, SubmitHandler, useForm } from 'react-hook-form'
+import { Select } from 'chakra-react-select'
+
+import {
+  chakra as Chakra,
+  Button,
+  Drawer,
+  DrawerBody,
+  DrawerCloseButton,
+  DrawerContent,
+  DrawerFooter,
+  DrawerHeader,
+  DrawerOverlay,
+  Flex,
+  FormControl,
+  FormHelperText,
+  FormLabel,
+  HStack,
+  Image,
+  Radio,
+  RadioGroup,
+  Text,
+  useDisclosure,
+  VStack,
+  AlertStatus,
+  useToast,
+} from '@chakra-ui/react'
+import {
+  AdapterExportError,
+  ExportFormat,
+  ExportFormatDisplay,
+  ProtocolAdapterTabIndex,
+} from '@/modules/ProtocolAdapters/types.ts'
+import useGetAdapterInfo from '@/modules/ProtocolAdapters/hooks/useGetAdapterInfo.ts'
+import { adapterExportFormats } from '@/modules/ProtocolAdapters/utils/export.utils.ts'
+import { DEFAULT_TOAST_OPTION } from '@/hooks/useEdgeToast/toast-utils.ts'
+
+interface SelectedExportFormat {
+  content: ExportFormat.Type
+  format: string
+}
+
+interface MIMETypeOptions {
+  value: string
+  label: string
+  description: string
+}
+
+const ExportDrawer: FC = () => {
+  const { t } = useTranslation()
+  const { isOpen, onOpen, onClose } = useDisclosure()
+  const toast = useToast()
+  const { adapterId } = useParams()
+  const navigate = useNavigate()
+  const { protocol, adapter } = useGetAdapterInfo(adapterId)
+  const { name, logoUrl } = protocol || {}
+  const [formatOptions, setFormatOptions] = useState<MIMETypeOptions[]>([])
+  const form = useForm<SelectedExportFormat>({
+    mode: 'all',
+    criteriaMode: 'all',
+    defaultValues: { content: ExportFormat.Type.CONFIGURATION },
+  })
+  const watchFormatChange = form.watch('content')
+
+  useEffect(() => {
+    if (adapterId) {
+      onOpen()
+    }
+  }, [adapterId, onOpen])
+
+  useEffect(() => {
+    const format = adapterExportFormats.find((e) => e.value === watchFormatChange)
+    if (format) {
+      const mimeOptions =
+        format.formats?.map<MIMETypeOptions>((format) => ({ label: format, value: format, description: '' })) || []
+      setFormatOptions(mimeOptions)
+      if (mimeOptions.length) form.setValue('format', mimeOptions[0].value)
+    }
+  }, [form, watchFormatChange])
+
+  const handleInstanceClose = () => {
+    onClose()
+    navigate('/protocol-adapters', { state: { protocolAdapterTabIndex: ProtocolAdapterTabIndex.PROTOCOLS } })
+  }
+
+  const handleEditorOnSubmit: SubmitHandler<SelectedExportFormat> = (data) => {
+    const downloader = adapterExportFormats.find((e) => e.value === data.content)
```

Comment:
suggestion: found an variable that might need some love

```suggestion
    const downloader = adapterExportFormats.find((exportFormat) => exportFormat.value === data.content)
```